codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2GoAwayFrame.java

```diff
@@ -27,6 +27,7 @@
 public final class DefaultHttp2GoAwayFrame extends DefaultByteBufHolder implements Http2GoAwayFrame {
     private final long errorCode;
     private int extraStreamIds;
+    private int lastStreamId = -1;
```

Comment:
consider cascading all constructors to the newly introduced constructor and just initializing `lastStreamId` in a single place.